hera_sim/simulate.py

```diff
@@ -31,226 +39,386 @@ def new_func(*args, **kwargs):
+        times: np.ndarray of float
+            Observed times in JD.
+        pols: list of str
+            Polarization strings.
+        red_grps: list of list of int
+            Redundant baseline groups. Each entry is a list containing the baseline
+            integer for each member of that redundant group.
+        red_vecs: list of np.ndarray of float
+            Average of all the baselines for each redundant group.
+        red_lengths: list of float
+            Length of each redundant baseline.
         """
+        # TODO: add ability for user to specify parameter names to look for on
+        # parsing call signature
         # create some utility dictionaries
         self._components = {}
-        self.extras = {}
         self._seeds = {}
         self._antpairpol_cache = {}
+        self._filter_cache = {
+            "delay": {},
+            "fringe": {},
+        }
 
         # apply and activate defaults if specified
         if defaults_config:
             self.apply_defaults(defaults_config)
 
         # actually initialize the UVData object stored in self.data
         self._initialize_data(data, **kwargs)
-
-    @property
+        self._calculate_reds(tol=redundancy_tol)
+        self.extras = self.data.extra_keywords
+        for param in ("Ntimes", "Nfreqs", "Nblts", "Npols", "Nbls"):
+            setattr(self, param, getattr(self.data, param))
+        self.Nants = len(self.antpos)
+        self.get_data = self.data.get_data
+        self.get_flags = self.data.get_flags
+
+    @cached_property
     def antpos(self):
         # TODO: docstring
         """
         """
         antpos, ants = self.data.get_ENU_antpos(pick_data_ants=True)
         return dict(zip(ants, antpos))
 
-    @property
+    @cached_property
     def lsts(self):
-        # TODO: docstring
-        return np.unique(self.data.lst_array)
+        """Observed Local Sidereal Times in radians."""
+        # This process retrieves the unique LSTs while respecting phase wraps.
+        unique_lsts, inverse_inds, counts = np.unique(
+            self.data.lst_array, return_inverse=True, return_counts=True
+        )
+        return unique_lsts[inverse_inds[:: counts[0]]]
 
-    @property
+    @cached_property
     def freqs(self):
         """Frequencies in GHz."""
         return np.unique(self.data.freq_array) / 1e9
 
-    @property
+    @cached_property
     def times(self):
-        """Return unique simulation times."""
+        """Simulation times in JD."""
         return np.unique(self.data.time_array)
 
+    @cached_property
+    def pols(self):
+        """Array of polarization strings."""
+        return self.data.get_pols()
+
     def apply_defaults(self, config, refresh=True):
-        # TODO: docstring
         """
+        Apply the provided default configuration.
+
+        Equivalent to calling ``hera_sim.defaults`` with the same parameters.
+        See ``hera_sim.defaults.set`` documentation for further details.
```

Comment:
use `:meth:`

Reply:
Done, though I'm not sure if the updated version is exactly right? I put `:meth:` before both references in this docstring.